identity_account_storage/src/storage/stronghold.rs

```diff
@@ -81,8 +84,9 @@ impl Storage for Stronghold {
 
     let public_key: PublicKey = retrieve_public_key(&tmp_client, &tmp_location)?;
 
-    let did: IotaDID = IotaDID::new_with_network(public_key.as_ref(), network)
-      .map_err(|err| crate::Error::DIDCreationError(err.to_string()))?;
+    let did: CoreDID = IotaDID::new_with_network(public_key.as_ref(), network)
+      .map_err(|err| crate::Error::DIDCreationError(err.to_string()))?
```

Comment:
This should match on the `DIDType` parameter, to cause a compilation error when it is extended in the future.